packages/spanner/src/database.js

```
@@ -598,11 +633,18 @@ Database.prototype.runStream = function(query, options) {
     session: this.formattedName_
   });
 
+  var fields = {};
+
   if (reqOpts.params) {
-    var fields = {};
+    reqOpts.types = reqOpts.types || {};
 
     for (var prop in reqOpts.params) {
       var field = reqOpts.params[prop];
+
+      if (!reqOpts.types[prop]) {
```

When that happened I had been setting the types to 'unspecified' and assumed that the upstream would handle it, but we can throw an exception too